modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilder.java

```diff
@@ -73,11 +79,36 @@ private void failIfHeaderDoesNotSupportMultipleValues(String headerName)
         }
     }
 
+    /**
+     *
+     * Check if the given header allows multiple values. In case it doesn't and the header already has a value,
+     * a warn message is shown and the older values are removed from the map.
+     *
+     * @param headerName the header name to check if it has already a value set.
+     */
+    private void logIfHeaderDoesNotSupportMultipleValues(String headerName)
+    {
+        if (uniqueHeadersNames.contains(headerName))
```

Comment:
Should we consider the case here? Just thinking about the several issues we've had around that.